ChatSecure/Classes/Model/Yap Storage/OTRBaseMessage.m

```diff
@@ -270,6 +285,36 @@ + (void)deleteAllMessagesForAccountId:(NSString *)uniqueAccountId transaction:(Y
     [[transaction ext:extensionName] enumerateEdgesWithName:edgeName destinationKey:uniqueAccountId collection:[OTRAccount collection] usingBlock:^(YapDatabaseRelationshipEdge *edge, BOOL *stop) {
         [self deleteAllMessagesForBuddyId:edge.sourceKey transaction:transaction];
     }];
+    
+    //#865
+    [self removeAccountFromGroups:uniqueAccountId transaction:transaction];
```

Comment:
This might add unexpected behavior. The current setting to delete messages on disconnection doesn't delete your roster, and rooms are sort of your roster. I think this should be extracted and discussed further in a separate issue.

Reply:
With this one, are you saying I should create a separate ticket for handling Groups when auto-deleting on disconnect or manually deleting a group thread? Or just shelf it for now and think about what the expected behavior should be?

Reply:
Yeah I think that should be separated into a different issue for further discussion so it doesn't hold up this PR.

Reply:
Looking at this again, I am going around in circles (and making a mess) as to the best way to handle this. The changes I started to make in YapDatabaseRelationshipTransaction would work for this ChatSecure situation but may not fit with their intentions for the code. Inside the YDB_DeleteDestinationIfSourceDeleted condition (and YDB_DeleteSourceIfDestinationDeleted), I was planning to check for the YDB_NotifyIfSourceDeleted condition and notify before deleting in order to allow preprocessing (removing the media item file at its path on disk). However, the NotifyIf conditions allow the listener to modify the object and send it back. That wouldn't do any good here because after the notification, we are deleting the object. If the client code modified it, it would then just get deleted. I'm probably not explaining that well.

I'm still not sure why delete and notify doesn't work the way it says at https://github.com/yapstudios/YapDatabase/wiki/Relationships#notify. 

One other option might be for us to leave YapDatabase alone (though I still think it isn't doing what its intended to do), and just put YDB_NotifyIfSourceDeleted for media items. This way, the notification works correctly and we can remove the file from disk, and then from that method call a thread to delete the mediaItem object. Thoughts?